src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
```

Comment:
This whole if block is wrong for digest auth. The server will send the client a server nonce. That nonce is usually an encoded value which the server can reverse and understand. Best practice is to have an expiry on the server provided nonce which can be decoded from the nonce as the server should change the nonce periodically. This means even if you have authenticated previously and have a valid prior challenge, the server can decide at any point that the previous nonce is no longer valid and will issue a new challenge and return an Unauthorized response. In that scenario, the HttpClient should recalculate the authorization header instead of returning an unauthorized response to the client. Returning unauthorized to the client means your credentials weren't accepted whereas in this scenario they are fine, the server has just started using a new nonce.